src/compiler/parser.ts

```diff
@@ -2450,10 +2450,27 @@ namespace ts {
             if (token === SyntaxKind.LessThanToken) {
                 return true;
             }
-
             return token === SyntaxKind.OpenParenToken && lookAhead(isUnambiguouslyStartOfFunctionType);
         }
 
+        function skipParameterStart(): boolean {
+            if (isModifierKind(token)) {
+                // Skip modifiers
+                parseModifiers();
+            }
+            if (isIdentifier()) {
+                nextToken();
+                return true;
+            }
+            if (token === SyntaxKind.OpenBracketToken || token === SyntaxKind.OpenBraceToken) {
+                // Return true if we can parse an array or object binding pattern with no errors
+                const count = parseDiagnostics.length;
+                parseIdentifierOrPattern();
+                return count === parseDiagnostics.length;

```

Comment:
We're already in a `lookAhead` call so no point in doing speculative parsing again. Also, `tryParse` doesn't check whether errors were generated.